lib/readline/promises.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const {
+  Promise,
+} = primordials;
+
+const {
+  Readline,
+} = require('internal/readline/promises');
+
+const {
+  Interface: _Interface,
+  kQuestionCancel,
+} = require('internal/readline/interface');
+
+const {
+  AbortError,
+} = require('internal/errors');
+
+class Interface extends _Interface {
+  // eslint-disable-next-line no-useless-constructor
+  constructor(input, output, completer, terminal) {
+    super(input, output, completer, terminal);
+  }
+  question(query, options = {}) {
+    return new Promise((resolve, reject) => {
+      if (options.signal) {
+        if (options.signal.aborted) {
+          return reject(new AbortError());
+        }
+
+        options.signal.addEventListener('abort', () => {
+          this[kQuestionCancel]();
+          reject(new AbortError());
+        }, { once: true });
+      }
+
+      super.question(query, resolve);
```

Comment:
This needs to remove the event listener added on `options.signal` (and should probably use `validateOptions`?) after super.question finished.